src/core/lib/surface/channel.cc

```diff
@@ -266,11 +266,21 @@ grpc_channel* grpc_channel_create(const char* target,
     }
   }
   grpc_channel_stack_builder_set_channel_arguments(builder, args);
-  grpc_channel_args_destroy(args);
   grpc_channel_stack_builder_set_target(builder, target);
   grpc_channel_stack_builder_set_transport(builder, optional_transport);
+  grpc_core::RefCountedPtr<grpc_core::ChannelStackModifier>
```

Comment:
I believe this can be done without modifying this file at all...

If we register a config builder that does something like:

```
builder->channel_init()->RegisterStage([](grpc_channel_stack_builder* b) {
  auto* modifier = GetModifierFromArgs(grpc_channel_stack_buildlder_get_channel_arguments(b));
  if (modifier == nullptr) return true;
  return (*modifier)(b);
}
```